packages/playwright-test/src/expect.ts

```diff
@@ -180,6 +196,38 @@ class ExpectMetaInfoProxyHandler {
   }
 }
 
+function pollMatcher(matcher: any, timeout: number, thisArg: any, generator: () => any, ...args: any[]) {
+  return new Promise(async resolve => {
```

Comment:
nit: why not just an async function here?

Reply:
fair. fixed!